addons/event_sms/models/event_mail.py

```diff
@@ -7,29 +7,35 @@
 class EventTypeMail(models.Model):
     _inherit = 'event.type.mail'
 
-    notification_type = fields.Selection(selection_add=[
-        ('sms', 'SMS')
-    ], ondelete={'sms': 'set default'})
-    sms_template_id = fields.Many2one(
-        'sms.template', string='SMS Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the SMS that will be automatically sent')
-
     @api.model
-    def _get_event_mail_fields_whitelist(self):
-        return super(EventTypeMail, self)._get_event_mail_fields_whitelist() + ['sms_template_id']
+    def _selection_template_model(self):
+        return super(EventTypeMail, self)._selection_template_model() + [('sms.template', 'SMS')]
+
+    notification_type = fields.Selection(selection_add=[('sms', 'SMS')], ondelete={'sms': 'set default'})
+
+    @api.depends('notification_type')
+    def _compute_notification_type_model_id(self):
+        sms_model = self.env['ir.model'].sudo().search([('model', '=', 'sms.template')], limit=1)
+        sms_mails = self.filtered(lambda mail: mail.notification_type == 'sms')
+        sms_mails.notification_type_model_id = sms_model
+        super(EventTypeMail, self - sms_mails)._compute_notification_type_model_id()
 
 
 class EventMailScheduler(models.Model):
     _inherit = 'event.mail'
 
-    notification_type = fields.Selection(selection_add=[
-        ('sms', 'SMS')
-    ], ondelete={'sms': 'set default'})
-    sms_template_id = fields.Many2one(
-        'sms.template', string='SMS Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the SMS that will be automatically sent')
+    @api.model
+    def _selection_template_model(self):
+        return super(EventMailScheduler, self)._selection_template_model() + [('sms.template', 'SMS')]
+
+    notification_type = fields.Selection(selection_add=[('sms', 'SMS')], ondelete={'sms': 'set default'})
```

Comment:
HumanLinter: unnecessary blank line above

Reply:
But here there's the method `_selection_template_model` just above (not a field) so it's better to have a blank line between the field and the method ?

Reply:
DId not see there were not any other field left. Obviously we keep a blank line in that case :D